src/libcore/intrinsics.rs

```diff
@@ -278,21 +278,199 @@ extern "rust-intrinsic" {
     /// Moves a value out of scope without running drop glue.
     pub fn forget<T>(_: T) -> ();
 
-    /// Unsafely transforms a value of one type into a value of another type.
+    /// Reinterprets the bits of a value of one type as another type. Both types
+    /// must have the same size. Neither the original, nor the result, may be an
+    /// [invalid value]
+    /// (https://doc.rust-lang.org/nomicon/meet-safe-and-unsafe.html).
     ///
-    /// Both types must have the same size.
+    /// `transmute::<T, U>(t)` is semantically equivalent to the following:
+    ///
+    /// ```
+    /// use std::{mem, ptr};
+    /// // assuming that T and U are the same size
+    /// unsafe fn transmute<T, U>(t: T) -> U {
+    ///     let mut u: U = mem::uninitialized();
+    ///     ptr::copy_nonoverlapping(&t as *const T as *const u8,
+    ///                              &mut u as *mut U as *mut u8,
+    ///                              mem::size_of::<T>());
+    ///     mem::forget(t);
+    ///     u
+    /// }
+    /// ```
+    ///
+    /// `transmute` is incredibly unsafe. There are a vast number of ways to
+    /// cause undefined behavior with this function. `transmute` should be
+    /// the absolute last resort.
```

Comment:
I think it would be good to have a sentence here which briefly describes why transmute is unsafe, and hopefully inspires the reader to go to the nomicon to find out more. We can't expect that every reader will also read the nomicon.

> We want to keep away people however possible

I don't agree. We want to give users enough information to make an informed choice and to properly understand the trade-offs they are making with that choice.